<launch>
    <arg name="map_path" description="path to commonroad or lanelet2 map"/>
    <arg name="solution_file" description="path to commonroad solution file"/>
    <arg name="vehicle_model" default="sample_vehicle" description="vehicle model"/>

    <let name="vehicle_description_package" value="$(eval '\'$(var vehicle_model)\' + \'_description\'')"/>
    <let name="vehicle_info_param_file" value="$(find-pkg-share $(var vehicle_description_package))/config/vehicle_info.param.yaml" />

    <node pkg="cr2autoware" exec="cr2autoware_node" name="cr2autoware_node">
        <param from="$(find-pkg-share cr2autoware)/param/cr2autoware_param_file.param.yaml"/>
        <param name="general.map_path" value="$(var map_path)"/>
        <param name="general.solution_file" value="$(var solution_file)"/>
        <param name="vehicle.vehicle_info_param_file" value="$(var vehicle_info_param_file)"/>
    </node>
</launch>
